test/index.js

```diff
@@ -639,6 +639,22 @@ describe('applyToDefaults()', () => {
         expect(merged).to.equal({ a: { b: 4 }, c: { g: { r: { h: 8 } } } });
     });
 
+    it('shallow copies the nested keys (undefined)', () => {
+
+        const defaults = {
+            a: {
+                b: 1
+            }
+        };
+
+        const source = {
+            a: undefined
+        };
+
+        const merged = Hoek.applyToDefaults(defaults, source, { shallow: ['a.b'] });
+        expect(merged).to.equal({ a: { b: 1 } });
```

Comment:
This is the correct value to expect when `nullOverride: false` (the default).

With `nullOverride: true`, I would expect it to equal `{ a: undefined }`.

FYI, this caused me to find another bug, where `applyToDefaults()` ignores `nullOverride: true` if _any_ array has been passed in the `shallow` option (even `shallow: []`).

Reply:
> FYI, this caused me to find another bug, where `applyToDefaults()` ignores `nullOverride: true` if any array has been passed in the `shallow` option (even `shallow: []`).

It looks that that functionality was lost in this commit: https://github.com/hapijs/hoek/commit/fdeebdba31314b5b570209e8d4bb70d02f75ac89#diff-92bbac9a308cd5fcf9db165841f2d90ce981baddcb2b1e26cfff170929af3bd1L209-L222

That commit was released in v7.0.0, though it doesn't appear it have been intentional.

I think this is better fixed in a separate PR, right?